pkg/cloudprovider/providers/azure/azure_storage.go

```diff
@@ -68,3 +68,19 @@ func (az *Cloud) DeleteFileShare(accountName, key, name string) error {
 	return nil
 
 }
+
+// ResizeFileShare resize a file share
+func (az *Cloud) ResizeFileShare(accountName, accountKey, name string, sizeGB int) error {
```

Comment:
cannot comment on unchanged lines but `createFileShare` looks buggy to me: in event of `SetProperties()` failure, the share should be deleted, otherwise a quota-less share is left unaccounted...

Reply:
`createFileShare` will delete share when hit `SetProperties()` failure, it's already implemented, see code:
https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/azure/azure_file.go#L46-L50